### PR TITLE
Takes the ABDUTOR_PROOF flag out back

### LINKED ISSUE #83844: Abductors can target the thunderdome and surrounding area
Reporting client version: 515.1640

## Round ID:
[230275](https://scrubby.melonmesa.com/round/230275)

## Testmerges:
- [[TM ONLY] Hijacked Revision Date for v3 TGS Api](https://www.github.com/tgstation/tgstation/pull/83227)

## Reproduction:

Spawn on a mothership
Set your species type to abductor
Interact with the cams console
Hit page down a bunch to jump down z levels until it stops updating
Scrub around static filled space until you find the thunderdome
You can teleport down :)

image

### PR DESCRIPTION
**Pull request by Rex9001**
## About The Pull Request
Basically the title, Abductors can no longer teleport into NOTELEPORT areas except for the abductor ship. This now also removes the ABDUCTOR_PROOF bitflag and replaces it with a check for ai monitored areas

fixes https://github.com/tgstation/tgstation/issues/83844
## Why It's Good For The Game
Abductors shouldnt kidnap people in centcom
## Changelog
:cl:
fix: Abductors can no longer teleport into Centcom
remove: Brutally murders the ABDUCTOR_PROOF flag
/:cl:

### CODE REVIEW

File: code/modules/antagonists/abductor/machinery/camera.dm [hunk 1]
```diff
@@ -65,6 +65,10 @@
 		to_chat(owner, span_warning("This area is too heavily shielded to safely transport to."))
 		return
 
+	if(target_area.area_flags & NOTELEPORT && !istype(target_area, /area/centcom/abductor_ship))
```
Comment (ZephyrTFA) [inline code/modules/antagonists/abductor/machinery/camera.dm:68]:
```suggestion
	if((target_area.area_flags & NOTELEPORT) && !istype(target_area, /area/centcom/abductor_ship))
```

**Review by ZephyrTFA — COMMENTED** [review]
why dont we just remove the abductor proof flag entirely if all its actually used for is to denote an area that abductors are not allowed to teleport into?

**Comment by Rex9001** [comment]
> why dont we just remove the abductor proof flag entirely if all its actually used for is to denote an area that abductors are not allowed to teleport into?

I wasnt sure about removing it because its used in one area, which is the AI sat where you are allowed to teleport normally but not as an abductor

**Comment by ZephyrTFA** [comment]
I'd prefer you make it so that abductors cant teleport into "critical" areas instead.

**Comment by ZephyrTFA** [comment]
Remove the snowflake flag

**Comment by Rex9001** [comment]
> I'd prefer you make it so that abductors cant teleport into "critical" areas instead.

I'll do it tomorrow, but please clarify here or on discord what you mean by critical areas and I'll fix this to that standard

**Review by ZephyrTFA — APPROVED** [review]

**Pull request merged**